services/vision/vision.go

```diff
@@ -203,6 +216,70 @@ func (vm *vizModel) GetObjectPointClouds(ctx context.Context, cameraName string,
 	return vm.segmenter3DFunc(ctx, cam)
 }
 
+func (vm *vizModel) CaptureAllFromCamera(
+	ctx context.Context,
+	cameraName string,
+	returnImage bool,
+	returnClass bool,
+	returnDet bool,
+	returnObjPCD bool,
+	extra map[string]interface{},
+) (viscapture.VisCapture, error) {
+	ctx, span := trace.StartSpan(ctx, "service::vision::ClassificationsFromCamera::"+vm.Named.Name().String())
+	defer span.End()
+	cam, err := camera.FromRobot(vm.r, cameraName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not find camera named %s", cameraName)
+	}
+	img, release, err := camera.ReadImage(ctx, cam)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get image from %s", cameraName)
+	}
+	defer release()
+	var detections []objectdetection.Detection
+	if returnDet {
+		if vm.detectorFunc == nil {
+			logger := vm.r.Logger()
+			logger.Debugf("detections requested but vision model %q does not implement a Detector", vm.Named.Name())
+		} else {
+			detections, err = vm.Detections(ctx, img, extra)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	var classifications classification.Classifications
+	if returnClass {
+		if vm.classifierFunc == nil {
+			logger := vm.r.Logger()
+			logger.Debugf("classifications requested in CaptureAll but vision model %q does not implement a Classifier",
+				vm.Named.Name())
+		} else {
+			classifications, err = vm.Classifications(ctx, img, 10, extra)
```

Comment:
Don't assume it will just be top 10 - set the number to 0 and return all classifications. OR, if that feels unacceptable, use the "extra" struct for this, by checking if extra contains a "top_n" key, and just convert that to an int.

Actually, I see that the "TopN" function in classifications.go has no way of returning all classifications. I think it should be edited so that n = 0 corresponds to returning all classifications